src/Setup_geometry.jl

```diff
@@ -447,6 +447,67 @@ function AddCylinder!(Phase, Temp, Grid::AbstractGeneralGrid;   # required input
     return nothing
 end
 
+
+
+function AddPolygon!(Phase, Temp, Grid::AbstractGeneralGrid;                 # required input
+    xlim=Tuple{}, ylim=Tuple{2}, zlim=Tuple{},     # limits of the box
```

Comment:
What are you trying to achieve with `Tuple{}` and `Tuple{2}`, this looks odd to me and probably not doing what you intend?

For example, you can't do `Tuple{2}[2]` as you below.